apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataReader.cs

```diff
@@ -254,6 +257,53 @@ public async Task<int> GetOrdinalAsync(string name, CancellationToken cancellati
             return _fieldIndex[name];
         }
 
+        /// <summary>
+        /// Gets the read timestamp if <see cref="TimestampBound.ReturnReadTimestamp" /> was true.
+        /// </summary>
+        /// <remarks>
+        /// The read timestamp is only available after at least one result is
+        /// read. If the returned results are empty, then a read timestamp
+        /// won't be available.
+        /// </remarks>
+        /// <returns>The value converted to a <see cref="Timestamp"/>.</returns>
+        /// <exception cref="InvalidOperationException">A read timestamp is unavailable,
+        /// possibly because <see cref="TimestampBound.ReturnReadTimestamp" /> was false.</exception>
+        public Timestamp GetReadTimestamp()
+        {
+            if (_readTimestamp != null)
+            {
+                return _readTimestamp;
+            }
+            return Task.Run(() => GetReadTimestampAsync(CancellationToken.None)).ResultWithUnwrappedExceptions();
```

Comment:
I worry a little about this. If I understand this correctly, anyone who calls this method __before__ having called `Read` or `ReadAsync`, will block until some other thread calls `Read` or `ReadAsync`. I don't think that is a good idea, as it could cause cause a thread to be blocked indefinitely, which can be very hard to debug, if someone accidentally calls this without having called `Read` first. I would rather just document clearly that:
1. `GetReadTimestamp` will only return a timestamp __after__ `Read` or `ReadAsync` has been called at least once for single-use transactions.
2. For multi-use transactions it will return it directly, as it is known from the read-only transaction that has already been started.

Reply:
I ran the integration test for single-use transactions to test this and it was happy to not have called `Read` or `ReadAsync`. Before I added the async variant, we were calling `PopulateMetadata()` which was already internally calling `PopulateMetadataAsync()` so the behavior didn't change much by making this method async.

Let me know if I misunderstood something.

Reply:
Interestingly, it even had the read timestamp without me calling `Read` or `ReadAsync` which makes me think that internally `ResultSet.GetMetadataAsync()` is somehow able to get the metadata anyway.

Reply:
Ah, yes, that seems to be the case for this client. The `ResultSet.GetMetaDataAsync()` method will call `ReliableStreamReader.GetMetaDataAsync()`, which again will call `ReliableStreamReader.EnsureInitializedAsync()`. The latter method automatically calls `MoveNextAsync()` (which is basically the same as `Read` or `ReadAsync`):
https://github.com/googleapis/google-cloud-dotnet/blob/13ed74b0e70f5b9932c9f679a3378493586c2fcc/apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ReliableStreamReader.cs#L170-L180